File: 02-java-spring-boot/src/main/java/com/tqi/checkout/service/EmailNotificadorService.java
package com.tqi.checkout.service;

import org.springframework.stereotype.Service;
// [🟥S] - SINGLE RESPONSIBILITY PRINCIPLE: Sua única razão para mudar é a regra de envio específica do canal de E-mails.
// COMPARAÇÃO COM JAVA: Implementa a interface 'Notificador' usando o operador ':'. 
// O Kotlin utiliza 'String Interpolation' nativa ($email, $valor) eliminando a concatenação manual com '+'.
@Service
public class EmailNotificadorService implements Notificador {
    @Override
    public void enviarComprovante(String email , double valor) {
        // KOTLIN REVOLUTION: String Interpolation nativa ($email, $valor).
        // Chega de concatenações confusas de strings com o operador '+'.
        System.out.println("E-mail enviado para "+ email +" confirmando os R$ "+ valor);
    }
}

File: 05-kotlin-puro/src/main/kotlin/com/dprev/checkout/modules/pagamento/service/EmailNotificadorService.kt
package com.dprev.checkout.modules.pagamento.service

//import org.springframework.stereotype.Service
// [🟥S] - SINGLE RESPONSIBILITY PRINCIPLE: Sua única razão para mudar é a regra de envio específica do canal de E-mails.
// COMPARAÇÃO COM JAVA: Implementa a interface 'Notificador' usando o operador ':'. 
// O Kotlin utiliza 'String Interpolation' nativa ($email, $valor) eliminando a concatenação manual com '+'.
//@Service
public class EmailNotificadorService : Notificador {
    public override fun enviarComprovante(destino: String, valor: Double) {
        // KOTLIN REVOLUTION: String Interpolation nativa ($email, $valor).
        val email = destino // <--- Aqui pegamos o argumento 'destino' que veio da interface
        println("E-mail enviado para $email confirmando os R$ $valor")
    }
}
